frontend/lib/parsing/parser-error-classes-list.cpp

```diff
@@ -653,4 +653,96 @@ void ErrorUnsupportedAsIdent::write(ErrorWriterBase& wr) const {
 
 /* end post-parse-checks errors */
 
+
+// main module / command line invocation errors
+void ErrorAmbiguousSourceFile::write(ErrorWriterBase& wr) const {
+  auto& loc = std::get<IdOrLocation>(info_);
+  auto& found = std::get<1>(info_);
+  auto& other = std::get<2>(info_);
+  auto& warnUnstable = std::get<3>(info_);
+  wr.heading(kind_, type_, loc,
+             "ambiguous module source file -- using ",
+             found,
+             " over ",
+             other);
+  if (warnUnstable) {
+    wr.note(loc, "which module is chosen in this case is unstable");
+  }
+}
+
+void ErrorAmbiguousMain::write(ErrorWriterBase& wr) const {
+  auto& loc = std::get<IdOrLocation>(info_);
+  auto& mainFns = std::get<std::vector<const uast::Function*>>(info_);
+  auto& modIds = std::get<std::vector<ID>>(info_);
+  auto& modNames = std::get<std::vector<UniqueString>>(info_);
+
+  bool allSameModule = true;
+  ID firstModuleId;
+  if (modIds.size() > 0) firstModuleId = modIds[0];
+  for (auto modId : modIds) {
+    if (modId != firstModuleId) {
+      allSameModule = false;
+    }
+  }
+
+  if (allSameModule) {
+    const char* moduleName = "<unknown>";
+    if (modNames.size() > 0)
+      moduleName = modNames[0].c_str();
+
+    wr.heading(kind_, type_, loc,
+               "multiple 'main' procedures in module '", moduleName, "'");
+    wr.message("");
+    for (auto fn : mainFns) {
+      wr.note(locationOnly(fn), "found 'proc main' here");
+      wr.codeForDef(fn);
+    }
+  } else {
+    wr.heading(kind_, type_, loc,
+               "unknown main module due to multiple 'main' procedures");
+    wr.note(loc, "please use '--main-module' to disambiguate");
+    wr.message("");
+    size_t i = 0;
+    for (auto fn : mainFns) {
+      const char* moduleName = "<unknown>";
+      if (i < modNames.size())
+        moduleName = modNames[i].c_str();
+      wr.note(locationOnly(fn),
+              "found 'proc main' in module '", moduleName, "'");
+      wr.codeForDef(fn);
+      i++;
+    }
+  }
+}
+void ErrorAmbiguousMainModule::write(ErrorWriterBase& wr) const {
+  auto& loc = std::get<IdOrLocation>(info_);
+  auto& modules = std::get<std::vector<const uast::Module*>>(info_);
+  wr.heading(kind_, type_, loc,
+             "unknown main module due to multiple command line modules");
+  wr.note(loc, "add a 'proc main' or use '--main-module'");
+  wr.message("");
+  size_t i = 0;
```

Comment:
I can't compile dyno after this PR due to this variable being set but unused. Not sure why CI didn't catch this, maybe a difference between it's clang 13 and my clang 15. But I think it should be removed, unless it was meant to be used somehow

Reply:
Thanks, sorry about that, I'll have a look.